compose/src/main/java/com/bottlerocketstudios/compose/navdrawer/NavDrawerItem.kt

```diff
@@ -0,0 +1,100 @@
+package com.bottlerocketstudios.compose.navdrawer
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.ColumnScope
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Icon
+import androidx.compose.material.LocalContentColor
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import com.bottlerocketstudios.compose.R
+import com.bottlerocketstudios.compose.resources.Dimens
+import com.bottlerocketstudios.compose.util.Preview
+import com.bottlerocketstudios.launchpad.compose.bold
+import kotlinx.coroutines.launch
+
+data class NavItemState(
+    val icon: Int,
+    val itemText: Int,
```

Comment:
@BottleRocket-Colin Good place to add the androidx annotations for drawable and string:

```suggestion
    @DrawableRes val icon: Int,
    @StringRes val itemText: Int,
```